graph/graph.go

```diff
@@ -27,11 +28,64 @@ import (
 	"github.com/docker/docker/runconfig"
 )
 
+type retainedLayers struct {
```

Comment:
What I gather from reading the code below is that the keys in the map are referenced image IDs and the values for each is a slice of IDs which hold the references.